chargebee_byte/client.py

```diff
@@ -8,8 +9,19 @@ def __init__(self, site, api_key):
         self.auth = requests.auth.HTTPBasicAuth(api_key, '')
         self.api_url = 'https://{}.chargebee.com/api/v2'.format(site)
 
-    def get_subscriptions(self, parameters=None):
+    def get_paginated_subscriptions(self, parameters=None):
         request = SubscriptionRequest(parameters)
         response = requests.get(self.api_url + request.path, auth=self.auth, params=request.data)
         response.raise_for_status()
         return response.json()
+
+    def get_all_subscriptions(self, parameters=None):
+        ret = self.get_paginated_subscriptions(parameters)
+        subscriptions = ret['list']
+
+        if 'next_offset' in ret:
+            new_parameters = deepcopy(parameters) or {}
+            new_parameters.update({'offset': ret['next_offset']})
+            subscriptions += self.get_all_subscriptions(new_parameters)
```

Comment:
I'm not a huge fan of recursion if it can be avoided - it'll cause some extra memory usage and cause stacktraces to increase in size if an exception happens inside of it. What do you think of something like:

```python
new_params = deepcopy(parameters) or {}
ret = {'next_offset': 0}  # fake response for initial request
subs = []
while 'next_offset' in ret:
    new_params['offset'] = ret['next_offset']
    ret = self.get_paginated_subscriptions(new_params)
    subs.extend(ret['list'])
return subs
```

Reply:
I've added it in, though I did like my recursive thingy :sob: Always when I write something recursive I get rid of it later :smile:

Reply:
P.s. I checked and it looks like specifying an offset of `''` works (as in, it doesn't break or regard it as an offset)